runtime/README.rst

```diff
@@ -3,35 +3,95 @@
-- `tests <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/tests>`_ : A collection of C++ tests for modules and methods in the runtime.
+- `include <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/include>`_:
+    This contains the public header files of the runtime including the ``QuantumDevice`` API
+    for backend quantum devices and the runtime CAPI.
+
+- `extensions <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/extensions>`_:
+    A collection of extensions for backend simulators to fit into the
+    `QIR programming model <https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/4_Quantum_Runtime.md#qubits>`_.
+    The `StateVectorDynamicCPU <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/extensions/StateVectorDynamicCPU.hpp>`_
+    class extends the state-vector class of `Pennylane-Lightning <https://github.com/PennyLaneAI/pennylane-lightning>`_ providing
+    dynamic allocation and deallocation of qubits.
+
+- `lib <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib>`_:
+    The core modules of the runtime are structured into ``lib/capi`` and ``lib/backend``.
+    `lib/capi <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib/capi>`_  implements the bridge between
+    QIR instructions in LLVM-IR and C++ device backends. `lib/backend <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/lib/backend>`_
+    contains implementations of the ``QuantumDevice`` API for backend simulators.
+
+- `tests <https://github.com/PennyLaneAI/catalyst/tree/main/runtime/tests>`_:
+    A collection of C++ tests for modules and methods in the runtime.
+
+Backend Devices
+===============
+
+New device backends for the runtime can be realized by implementing the quantum device interface.
+The following table shows the available devices along with supported features:
+
+.. list-table::
+   :widths: 25 25 25
+   :header-rows: 0
+
+   * - **Features**
+     - **PennyLane-Lightning**
+     - **PennyLane-Lightning-Kokkos**
+   * - Qubit Management
+     - Dynamic allocation/deallocation
+     - Static allocation/deallocation
+   * - Gate Operations
+     - `Lightning operations <https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/src/gates/GateOperation.hpp>`_
+     - `Lightning operations <https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/src/gates/GateOperation.hpp>`_
+   * - Quantum Observables
+     - ``Identity``, ``PauliX``, ``PauliY``, ``PauliZ``, ``Hadamard``, ``Hermitian``, ``Hamiltonian``, Tensor Product of Observables
+     - ``Identity``, ``PauliX``, ``PauliY``, ``PauliZ``, ``Hadamard``, ``Hermitian``, ``Hamiltonian``, Tensor Product of Observables
+   * - Expectation Value
+     - All observables
+     - All observables
+   * - Variance
+     - ``Identity``, ``PauliX``, ``PauliY``, ``PauliZ``
+     - Not supported
+   * - Finite Shots (``Sample``)
```

Comment:
I think this category should just be called "sampling", as sampling can only ever be finite.
Another category for finite shots / shot noise in reference to the computation of expectation values / variances / probabilities should be added to indicate that the backends do not support the latter.

```suggestion
   * - Sampling
```

Reply:
Fair points! I'll update the list.